### DISCUSSION TITLE
New member write permission

### DISCUSSION THREAD

**Discussion by ippocratis** [J-Rios/TLG_JoinCaptchaBot#174] [Ideas]
New member can post messages in the chat until their time to solve th captcha is out
Then the bot deletes the messages if no answer is given or an incorrect one is given and gets kicked
That makes perfect sense for the captcha puzzles as he has to import the solution somehow
But in my use case that I use a poll it would make more sense if he couldn't post no message at all until he provides a correct answer
I dont know if something like that is beyond the scope of the app but I'd love to see that implemented

**Comment by J-Rios** [comment]
Hello,

I am aware of that and I want to modify the behavior of the captcha modes "poll" and "button" to not allow new members to write any message as long as they have not solved the captcha. It hasn't been changed yet because I haven't had time available to do it, but it will surely be changed in the near future.

I have just created a new issue reagrding this request to have it written down:

https://github.com/J-Rios/TLG_JoinCaptchaBot/issues/175

Regards.

**Comment by J-Rios** [comment]
Update: Request implemented and applied.

https://github.com/J-Rios/TLG_JoinCaptchaBot/commit/3bc73e3ddf53ccecc3371a69da81126237267679

Best Regards.

**Reply by ippocratis** [reply]
Tested it on my group with a test acount
works great


Thanks and keep up

**Reply by J-Rios** [reply]
Nice to hear!
Thank you for testing :)
